public/js/components/FormFields/FormFieldArrayWrapper.js

```diff
@@ -18,27 +18,51 @@ export default class FormFieldArrayWrapper extends React.Component {
     ])
   }
 
+  //We store new items in state to avoid sending invalid items back to the server before they're ready
+  state = {
+    newItems: []
+  }
+
   onAddClick = () => {
-    const existingValues = this.props.fieldValue || [];
-    this.props.onUpdateField(existingValues.concat([undefined]));
+    this.setState({
+      newItems: this.state.newItems.concat([undefined])
```

Comment:
Why the `undefined`? 😮

Reply:
I see, it's like a placeholder for future values.

Reply:
Yep, and this was the cause of the problems before - it was being immediately propagated back up and sent to the server.
Interestingly, we accidentally worked around this for e.g. timeline because the child Scribe component immediately updates the props and overwrites the previous prop update (the one with `undefined`), before it can be sent to the server